### ISSUE TITLE
S3: can't download files uploaded with gzip content-encoding 

### ISSUE THREAD

**Issue by butla** [localstack/localstack#722]
**Steps to reproduce:**
1. run `docker-compose up` with `docker-compose.yml` attached below.
2. Create a bucket called "test-bucket"
3. Run `test_localstack_s3.sh`, attached below.

**Excpected behavior:**
The file gets downloaded as `downloaded_bla.json.gz`.

**Actual behavior:**
AWS CLI freezes for some time and then produces `download failed: s3://test-bucket/bla.json.gz to ./downloaded_bla.json.gz Max Retries Exceeded`. No logs are produced by the `localstack` container during this event.

**docker-compose.yml:**
```
---
version: '2'
services:
  aws:
    image: localstack/localstack:0.8.6
    ports:
      - "4572:4572"
    environment:
      - SERVICES=s3
```

**test_localstack_s3.sh:**
```
PORT=4572
BUCKET=test-bucket
set -v

echo '{"a": 123, "b": "qwert"}' > bla.json
gzip bla.json
aws s3 --endpoint-url=http://localhost:${PORT} cp \
    --content-type application/json --content-encoding gzip \
    bla.json.gz s3://${BUCKET}/
aws s3 --endpoint-url=http://localhost:${PORT} \
    cp s3://${BUCKET}/bla.json.gz downloaded_bla.json.gz
```

**More context:**
If I just ommit the `--content-encoding gzip` then everything works properly.

I saw in one issue report, that localstack has a `--debug` option, so I wanted to run it with that. But I couldn't install it with pip due to errors.

I also tried rebuilding the image so that it would start with the debug option, which I guess can be done by modifying the parameters (adding `--debug`) in the "infra" target in the Makefile, but I got errors when trying to build the image...

**Comment by mvayngrib** [comment]
experiencing the same issue. This used to work :(

**Comment by whummer** [comment]
Thanks for reporting @butla. This should be fixed in #920 - please re-open this issue if the problem persists
